File: 1781-sum-of-beauty-of-all-substrings/1781-sum-of-beauty-of-all-substrings.js
/**
 * @param {string} s
 * @return {number}
 */
var beautySum = function(s) {
  const n = s.length;
  let ans = 0;
  
  for (let i = 0; i < n; i++) {
    const freq = new Array(26).fill(0);
    let max = 0;
    for (let j = i; j < n; j++) {
      max = Math.max(max, ++freq[s.charCodeAt(j) - 'a'.charCodeAt()]);
      let min = Math.min(...freq.filter((x) => x));
      ans += max - min;
    }
  }
  
  return ans;
};

File: 1781-sum-of-beauty-of-all-substrings/1781-sum-of-beauty-of-all-substrings.py
class Solution:
    def beautySum(self, s: str) -> int:
        n, ans = len(s), 0
        
        for i in range(n):
            counter = Counter()
            for j in range(i, n):
                counter[s[j]] += 1
                ans += max(counter.values()) - min(counter.values())
        return ans